src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const object = sourceString
+    .split(';')
+    .map((element) => {
+      return element.split(':').map((part) => part.trim());
+    })
+    .filter((array) => array.length !== 1)
+    .reduce((result, element) => {
+      result[element[0]] = element[1];
+
+      return result;
+    }, {});
+
+  return object;
```

Comment:
The chained methods should be indented at the same level to improve readability, as per the checklist's good example.